matrix/matrix.py

```diff
@@ -19,21 +32,21 @@ def _validate(self, data: tuple):
         """Check if the data given is valid.
 
         note:
-            Valid data is defined if it contains tuples of the same length
-            (Matrix only supports n * n), items can't be iterable.
+            Valid data is defined if it contains tuples of the same length,
+            Supports only squared matrices.
         """
-        # if any(hasattr(item, '__len__') for line in data for item in line):
-        #     raise TypeError("Matrix item type not supported")
-
         if not all([len(line) == len(data) for line in data]):
             raise ValueError("Matrix must be even-sided")
 
     @classmethod
     def unity(cls, dimension: int):
         """Generate a matrix with a main diagonal of 1's, elsewhere 0's."""
-        return cls(
-            tuple(tuple((0,) * i + (1,) + (0,) * (dimension - i - 1))
-                  for i in range(dimension)))
+        unity_matrix = []
+
+        for i in range(dimension):
+            unity_matrix.append((0,) * i + (1,) + (0,) * (dimension - i - 1))
```

Comment:
No need to save it to list and then convert to tuple, save it from the start to tuple